apps/hash-api/src/graph/ontology/primitive/util.ts

```diff
@@ -14,6 +16,14 @@ import { ImpureGraphFunction } from "../../context-types";
 import { getOrgById } from "../../knowledge/system-types/org";
 import { getUserById } from "../../knowledge/system-types/user";
 
+export const isExternalTypeId = (typeId: VersionedUrl) =>
+  !typeId.startsWith(frontendUrl) &&
+  // To be removed in H-1172: Temporary provision to serve types with a https://hash.ai URL from https://app.hash.ai
+  !(
+    ["https://app.hash.ai", "http://localhost:3000"].includes(frontendUrl) &&
+    new URL(typeId).hostname === "hash.ai"
+  );
```

Comment:
Data types and property types can't be created at the moment in `main` because this patch doesn't exist